lib/ansible/modules/cloud/vmware/vmware_guest_instantclone.py

```diff
@@ -0,0 +1,265 @@
+'''
+
+HAS_PYVMOMI = False
+try:
+    from pyVmomi import vim, vmodl, VmomiSupport
+    HAS_PYVMOMI = True
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.common.network import is_mac
+from ansible.module_utils._text import to_text, to_native
+from ansible.module_utils.vmware import (find_obj, gather_vm_facts, vmware_argument_spec,
+                                         set_vm_power_state, PyVmomi, wait_for_task, TaskError)
+
+
+class PyVmomiHelper(PyVmomi):
+    def __init__(self, module):
+        super(PyVmomiHelper, self).__init__(module)
+
+    def deploy_instantclone(self, vm):
+        vm_state = gather_vm_facts(self.content, vm)
+        # An instant clone requires the base VM to be running, fail if it is not
+        if vm_state['hw_power_status'] != 'poweredOn':
+            self.module.fail_json(msg='Unable to instant clone a VM in a "%s" state. It must be powered on.' % vm_state['hw_power_status'])
+        if vm_state['instant_clone_frozen'] is None:
+            self.module.fail_json(msg='Unable to determine if VM is in a frozen state. Is vSphere running 6.7 or later?')
+
+        # Build VM config dict
+        config = []
+        # If customvalues is not empty, fill config
+        if len(self.module.params['customvalues']) != 0:
+            for kv in self.module.params['customvalues']:
+                if 'key' not in kv or 'value' not in kv:
+                    self.module.exit_json(msg="The parameter customvalues items required both 'key' and 'value' fields.")
+                ov = vim.OptionValue()
+                ov.key = kv['key']
+                ov.value = kv['value']
+                config.append(ov)
+
+        # Begin building the spec
+        instantclone_spec = vim.VirtualMachineInstantCloneSpec()
+        location_spec = vim.VirtualMachineRelocateSpec()
+        instantclone_spec.config = config
+        instantclone_spec.name = self.module.params['clone_name']
+
+        if vm_state['instant_clone_frozen'] is False:
+            # VM is not frozen, need to do prep work for instant clone
+            vm_network_adapters = []
+            devices = vm.config.hardware.device
+            for device in devices:
+                if isinstance(device, vim.VirtualEthernetCard):
+                    vm_network_adapters.append(device)
+
+            for vm_network_adapter in vm_network_adapters:
+                # Standard network switch
+                if isinstance(vm_network_adapter.backing, vim.VirtualEthernetCardNetworkBackingInfo):
+                    network_id = vm_network_adapter.backing.network
+                    device_spec = vim.VirtualDeviceConfigSpec()
+                    device_spec.operation = 'edit'
+                    device_spec.device = vm_network_adapter
+                    device_spec.device.backing = vim.VirtualEthernetCardNetworkBackingInfo()
+                    device_spec.device.backing.deviceName = network_id
```

Comment:
`network_id` is a `vim.Network` instance, so you should actually set `network_id.name` in `deviceName`.